test/e2e_node/e2e_service.go

```diff
@@ -208,6 +202,45 @@ func (es *e2eService) startServer(cmd *healthCheckCommand) error {
 	return fmt.Errorf("Timeout waiting for service %s", cmd)
 }
 
+func (es *e2eService) stopService(name string) error {
+	// Since node services are started with sudo, we must get the PID of the child process.
+	out, err := exec.Command("pidof", name).Output()
```

Comment:
Running multiple processes is most likely going to result in weird issues. 
I think the right fix here is to run get the pid from the go exec process and then run `kill -11 -<pid>`, followed by a `kill -9 -<pid>`. The `-` prefix will kill all child processes as well.